EventsExpress/ClientApp/src/services/EventsExpressService.js

```diff
@@ -293,14 +294,40 @@ export default class EventsExpressService {
             }),
         });
 
-        return !res.ok
-            ? {
-                error: {
-                    ErrorCode: res.status,
-                    massage: await res.statusText
-                }
+        if (res.ok) {
+            return await res.json();
+        }
+        else if (res.status === 401) {
+            return this.RefreshHandler(this.getResource, res, url);
```

Comment:
Is there a reason to use recursive calls? what would be the anchor of recursion in this case?

Reply:
refactored